activerecord/test/cases/associations/has_many_associations_test.rb

```diff
@@ -2021,7 +2029,7 @@ def test_ids_reader_cache_should_be_cleared_when_collection_is_deleted
 
   def test_zero_counter_cache_usage_on_unloaded_association
     car = Car.create!(name: "My AppliCar")
-    assert_no_queries do
+    assert_no_queries(ignore_none: false) do
```

Comment:
I'm suspecting this test to be flakey. It does a `SHOW FULL FIELD FROM` query, tried to revert all the change from this PR and it still fails pretty consistently. I was able to reproduce the failure on master but it's not as frequent, not why